monai/data/image_reader.py

```diff
@@ -672,6 +672,11 @@ def read(self, data: Union[Sequence[str], str, np.ndarray], **kwargs):
             data: file name or a list of file names to read.
 
         """
+        if (self.reader_lib == "openslide") and (not has_osl):
```

Comment:
could have these logics even earlier for example between line 646 and 652?

Reply:
Actually no, Nic is instantiating this class in LoadImage, so the error shouldn't be raised in `__init__` (it used to be raise without having `has_osl` and `has_cux`).  The first place that it can be raised here.

I agree that the best place would be when the class is instantiating but for that purpose `LoadImage` need to be changed: https://github.com/Project-MONAI/MONAI/blob/6bc94b52572745534897cba4339e12aa18030dfa/monai/transforms/io/array.py#L93

Reply:
sure you're right! 

(@Nic-Ma I'm approving this for now, perhaps we need another iteration to revise the base API?)